cpp/src/arrow/compute/kernels/scalar_nested_test.cc

```diff
@@ -116,6 +117,97 @@ TEST(TestScalarNested, ListElementInvalid) {
               Raises(StatusCode::Invalid));
 }
 
+void CheckListSlice(std::shared_ptr<Array> input, std::shared_ptr<Array> expected,
+                    const ListSliceOptions& args) {
+  ASSERT_OK_AND_ASSIGN(auto result, CallFunction("list_slice", {input}, &args));
+  ASSERT_EQ(result, expected);
```

Comment:
Can we use the standard helper function for this instead (`CheckScalarUnary`)? It'll check slices and such (but doesn't check the case when nested arrays have their own offset, so we'll still need separate tests for that)

Reply:
The discrepancies are probably because it'll slice the input to check that the function works with sliced arrays

Reply:
Ah how right you are, thank you. Joris also helped me figure this one out today. Good thing to use `CheckScalarUnary`. :+1: